stograde/specs/util.py

```diff
@@ -12,4 +15,22 @@ def check_dependencies(spec):
         try:
             os.stat(filepath)
         except FileNotFoundError:
-            warning('spec {}: required file "{}" could not be found'.format(spec['assignment'], filepath))
+            logging.warning('spec {}: required file "{}" could not be found'.format(spec['assignment'], filepath))
+
+
+def check_architecture(assignment, spec, ci):
+    # get check_architecture()
+    _, arch, _ = run(['uname', '-m'])
+    try:
+        if spec['architecture'] != arch:
+            if ci:
+                logging.info('Skipping {}: wrong architecture'.format(assignment))
+            else:
+                print('{} requires {} architecture. You have {}'
+                      .format(assignment, spec['architecture'], arch),
+                      file=sys.stderr)
+            return False
+        else:
+            return True
+    except KeyError:
+        return True
```

Comment:
What do you think about

```python
def check_architecture(assignment, spec, ci):
    # get check_architecture()
    _, arch, _ = run(['uname', '-m'])

    if spec.get('architecture', None) == arch:
        return True
    else:
        if ci:
            logging.info('Skipping {}: wrong architecture'.format(assignment))
        else:
            print('{} requires {} architecture. You have {}'
                  .format(assignment, spec['architecture'], arch),
                  file=sys.stderr)
        return False
```

That is, using dict.get with a default value instead of checking for KeyError?